### ISSUE TITLE
Add {episode} for episode numbers in file names, to avoid conflicts in Kodi's episode number scraping

### ISSUE THREAD

**Issue by AshleyAshes** [meeb/tubesync#497]
I've been discussing this here on the Kodi forums
https://forum.kodi.tv/showthread.php?tid=377301

In short, Kodi prioritizes the filename text as it attempts to figure out the episode numbers, and the video titles of some tech videos contain number sequences that it thus prefers.

An example is "Phil's Computer Lab 2014-10-08 Roland SC-55 vs AWE64 Gold Descent [1080p WEB-DL].mkv", which has a nice NFO file that says it's Season 2014, Episode 49, Kodi prefers to read the 'AWE64' as 'Season 1 Episode 64' and it gives that authority.

You can add {yyyy_mm_dd} but that doesn't seem to quite work.  If there was an {episode} that resulted in the episode number, one could then do 'S{yyyy}E{episode}' which would result in a filename output of 'S2014E49', or such, which is far more 'Kodi Friendly'.

Otherwise, I think I'll have to modify things and put the full episode titles within square brackets on output, as I think Kodi will then ignore that.  ...Maybe?  I'm testing more things.

Sorry for all these, this is actually a neat piece of software, I just keep hitting compatibility walls.

**Comment by meeb** [comment]
There are legacy issues discussing this at quite some length, for example https://github.com/meeb/tubesync/issues/60

This a lot more hassle than it might initially seem. It's on the list, but not implemented yet.

**Comment by AshleyAshes** [comment]
I understand and that is unfortunate.  If you review the thread I linked, apparently the fact that Kodi was able to scrap files offline NFO with only dates as their numbering, into "Season And Episode Numbers" is unintentional.  The devs said that shouldn't even work, even if it is working for me but only with 'a few issues' like from other posts.  It shouldn't work at all by their design.  So as it stands, without getting episode numbers, I think TubeSync won't work as a PVR for Kodi until that is addressed. Their design is rather firmly dependent on 'Season/Episode' numbers in file names.  (Or at least it is supposed to be.)

**Comment by meeb** [comment]
Yep, I've reviewed the Kodi requirements for this previously. Does `{yyyymmdd}` work for episode? Without the underscores? It's at least a "number" and increments in the correct order. Obviously "episodes" in the 20+ million range might cause other issues.

**Comment by AshleyAshes** [comment]
I'm actually unsure, and will have to research some more.  But it seems that Kodi has a 'very strong preference' for the S00E00 format, having a sorta hierarchy in which is uses formats for hits.  So if anything in a Video Title can be mistaken for that format, it prefers that.  It doesn't even have like a 'Read Left To right' for matching, just goes through the whole title.  Kodi doesn't seem to have any way to 'remark out' anything in a file name either.  For the longest time I thought square brackets accomplished that but I guess I imagined that.

Right now I'm mainly using a workaround to just omit 'YouTube Video Title' entirely, so just "Series Name 0000-00-00.ext" but now I'm running into a new issue: When there's multiple videos uploaded on the same date.  Seems Phil was a busy guy in 2014 and 2015, and this is resulting some download errors as two entries want the same file name

I'm now trying this again attaching the {key} value into the file name since that's unique per video, but I bet you that's gonna result in a bunch of file names with random sequences that occasionally get read as S00E00 or at least E00 format which it prefers over date in a file name.

**Comment by AshleyAshes** [comment]
Yup, I was right.  Even something like 'Phil's Computer Lab 2024-04-05 [1080p WEB-DL][EKvLsiE2-ng].mkv' gets mistaken as Season 1 Episode 2, because it has 'E2' in the file name.

I have used Kodi for 10 years but never realized how utterly dependent it is on the S00E00 formatting to avoid literally anything else being mistaken for an episode number.  So 'dates' work, but only if nothing else confuses it, and I thought I'd settled on something for that, but that blew up when there were multiple videos on the same date because now I have identical filenames.  If I could include the video title, that'd prevent the names, but video can contain numbering sequences in the title so that breaks that.

I noticed some issues when doing Lazy Game Reviews through SickChill, I do this show manually since LGR's channel is maintained on TheTVDB.  Even using the 0x00 format for LGR videos could result in episode titles being confused for numbers.  I specifically had to use '(Series Name) S00E00 (Video Title).ext' to eliminate all risk of scraping errors.

It seems I've picked the one YouTube channel to pull out all of the edge cases in using TubeSync with Kodi, it's kinda amazing, ha ha.

**Comment by meeb** [comment]
Tried Jellyfin? :grinning:

**Comment by AshleyAshes** [comment]
Well, for now, TinyMediaManager can use the NFOs as the basis to renumber the files to S0000E00 format, so that at least solves the issue.  While TubeSync is not prefect, I def find using it a fair bit easier than some of the other options.
